Loop/View Models/BolusEntryViewModel.swift

```diff
@@ -111,6 +118,12 @@ final class BolusEntryViewModel: ObservableObject {
             .receive(on: DispatchQueue.main)
             .sink { [weak self] _ in self?.update() }
             .store(in: &cancellables)
+        
+        // Look for Loop starting, and assume that if it started after assertCurrentPumpData that we want to call pumpManagerRefreshCompletion
+        NotificationCenter.default
+            .publisher(for: .LoopRunning, object: dataManager.loopManager)
```

Comment:
Isn't the observeLoopUpdates enough to detect the arrival of fresh pump data?

Reply:
I thought it did, but not quite.  The above listener (sink) watches for `.LoopDataUpdated` which is called a lot more times than when the pump manage signals it has been refreshed (`pumpManagerRecommendsLoop`).  I wasn't thrilled with all of those updates.